### ISSUE TITLE
Pretrained models for ResNet in SO(2) and SE(3)

### ISSUE THREAD

**Issue by olayasturias** [QUVA-Lab/escnn#73]
Hello all,

I wonder if anyone has a pretrained model available for the equivariant ResNet from [this example](https://github.com/QUVA-Lab/escnn/blob/master/examples/e2wrn.py) and the SE(3) equivariant model in [this other example](https://github.com/QUVA-Lab/escnn/blob/master/examples/se3_3Dcnn.py), using a bigger dataset like Imagenet or similar.

Thanks!

**Comment by sibasmarak** [comment]
Hi, you do not need to retrain an equivariant version of ResNet (or other large pretrained models) to obtain pretrained equivariant ResNet, rather you can "adapt" a pretrained ResNet to be equivariant to a certain group with architecture-agnostic equivariance methods, such as canonicalization.

Please feel free to check out: [Equivariant Adaptation of Large Pretrained Model](https://arxiv.org/pdf/2310.01647.pdf), NeurIPS 2023.

**Comment by olayasturias** [comment]
Hi Siba,
Thank you for your answer. That's a fascinating work!
From what I understood, instead of training an equivariant Resnet from scratch, you preceded a ResNet50 with your canonicalization network and then fine-tuned the ResNet while training that canonicalization module. Is that correct?
Do you have code examples of how did you make that work? I'm particularly interested in the network you implemented with the escnn library.
Is it similar to the CNN under [this notebook](https://github.com/QUVA-Lab/escnn/blob/master/examples/model.ipynb)? How many layers - and in general, which hyperparameters- suited you well?

**Comment by sibasmarak** [comment]
Hi, thank you for taking a look at the paper!
Yes, indeed. Note that you don't need to fine-tune per se (as we show in the case of the Segment-Anything Model, you can only train the equivariant canonicalization network to learn the identity orientation with prior regularization), and you would need a regularization loss to align the outputs of the canonicalization network and (pre-trained) dataset orientation.

We are planning a release of our user-friendly library before the end of February. We are adding examples and tutorials for people to get started with canonicalization. I will let you know once we release the library. A schematic of the pipeline is described in Figure 2. 

Yes, the canonicalization networks are similar to the notebook you have linked. We give a small detail of hyperparameter tuning in Appendix Section B. We tune for different values of the number of layers, kernel sizes, dropout (switching off dropout generally helped), and learning rates. Anyways, the canonicalization networks are extensively small compared to the actual pretrained model under consideration, which makes it lucrative (some parameter sizes are highlighted in Table 3).

**Comment by dmklee** [comment]
I pretrained some equivariant ResNets on ImageNet-1k.  The models and weights can be found [here](https://github.com/dmklee/equivision).  

The canonicalization approach is appealing since it can be applied to any pre-trained method.  I haven't had a chance to compare against it yet, but I'm curious if there is any performance gap.
